### ISSUE TITLE
How to dynamically update the settings panel

### ISSUE THREAD

**Issue by zhw2590582** [zhw2590582/ArtPlayer#397]
https://github.com/zhw2590582/ArtPlayer/issues/396

@5H4D0WILA

I created an issue for you, and I think you found a bug, I will fix it in the next version

I wrote a demo for you, you can see if it works: [demo](https://artplayer.org/?libs=&code=var%20art%20%3D%20new%20Artplayer(%7B%0A%09container%3A%20%27.artplayer-app%27%2C%0A%09url%3A%20%27%2Fassets%2Fsample%2Fvideo.mp4%27%2C%0A%09setting%3A%20true%2C%0A%09settings%3A%20%5B%0A%09%09%7B%0A%09%09%09width%3A%20200%2C%0A%09%09%09html%3A%20%27Subtitles%27%2C%0A%09%09%09tooltip%3A%20%27gogoanime01%27%2C%0A%09%09%09selector%3A%20%5B%0A%09%09%09%09%7B%0A%09%09%09%09%09default%3A%20true%2C%0A%09%09%09%09%09html%3A%20%27gogoanime01%27%2C%0A%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%7D%2C%0A%09%09%09%09%7B%0A%09%09%09%09%09html%3A%20%27gogoanime02%27%2C%0A%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%7D%2C%0A%09%09%09%5D%2C%0A%09%09%09onSelect%3A%20function%20(item)%20%7B%0A%09%09%09%09art.subtitle.switch(item.url%2C%20%7B%0A%09%09%09%09%09name%3A%20item.html%2C%0A%09%09%09%09%7D)%3B%0A%09%09%09%09return%20item.html%3B%0A%09%09%09%7D%2C%0A%09%09%7D%2C%0A%09%09%7B%0A%09%09%09width%3A%20200%2C%0A%09%09%09html%3A%20%27Provider%27%2C%0A%09%09%09tooltip%3A%20%27gogoanime%27%2C%0A%09%09%09selector%3A%20%5B%0A%09%09%09%09%7B%0A%09%09%09%09%09default%3A%20true%2C%0A%09%09%09%09%09html%3A%20%27gogoanime%27%2C%0A%09%09%09%09%7D%2C%0A%09%09%09%09%7B%0A%09%09%09%09%09html%3A%20%27zoro%27%2C%0A%09%09%09%09%7D%2C%0A%09%09%09%5D%2C%0A%09%09%09onSelect%3A%20function%20(item)%20%7B%0A%09%09%09%09if%20(item.html%20%3D%3D%3D%20%27gogoanime%27)%20%7B%0A%09%09%09%09%09art.setting.option%5B0%5D.tooltip%20%3D%20%27gogoanime01%27%3B%0A%09%09%09%09%09art.setting.option%5B0%5D.selector%20%3D%20%5B%0A%09%09%09%09%09%09%7B%0A%09%09%09%09%09%09%09default%3A%20true%2C%0A%09%09%09%09%09%09%09html%3A%20%27gogoanime01%27%2C%0A%09%09%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%09%7B%0A%09%09%09%09%09%09%09html%3A%20%27gogoanime02%27%2C%0A%09%09%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%5D%0A%09%09%09%09%7D%0A%0A%09%09%09%09if%20(item.html%20%3D%3D%3D%20%27zoro%27)%20%7B%0A%09%09%09%09%09art.setting.option%5B0%5D.tooltip%20%3D%20%27zoro01%27%3B%0A%09%09%09%09%09art.setting.option%5B0%5D.selector%20%3D%20%5B%0A%09%09%09%09%09%09%7B%0A%09%09%09%09%09%09%09default%3A%20true%2C%0A%09%09%09%09%09%09%09html%3A%20%27zoro01%27%2C%0A%09%09%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%09%7B%0A%09%09%09%09%09%09%09html%3A%20%27zoro02%27%2C%0A%09%09%09%09%09%09%09url%3A%20%27%2Fassets%2Fsample%2Fsubtitle.srt%27%2C%0A%09%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%5D%0A%09%09%09%09%7D%0A%0A%09%09%09%09%2F%2F%20Important%0A%09%09%09%09art.setting.update()%3B%0A%0A%09%09%09%09return%20item.html%3B%0A%09%09%09%7D%2C%0A%09%09%7D%2C%0A%09%5D%2C%0A%7D)%3B)

In simple terms, `art.setting.option` can access all setting options array, you can modify these options at will, but after modification, you must call the method `art.setting.update()` to dynamically update

**Comment by zhw2590582** [comment]
The latest version has been released: `artplayer@4.5.8`

**Comment by 5H4D0WILA** [comment]
ahh, i see
thanks for creating a new issue
ill update in an hour or so and let you know if everything works fine

**Comment by 5H4D0WILA** [comment]
update: 
the function doesnt work on my end

using art.setting.update(); causes the error "art is undefined"
and 
using artPlayer.setting.update(); causes the error "artPlayer.setting.update is not a function" (artPlayer is a variable that gets set to art once art is defined)

**Comment by zhw2590582** [comment]
You need to update to: `artplayer@4.5.8`

**Comment by 5H4D0WILA** [comment]
i did

**Comment by 5H4D0WILA** [comment]
lemme push my latest code

**Comment by 5H4D0WILA** [comment]
welp, looks like my website just broke
